kafka-client-test/src/nativeMain/kotlin/com/icemachined/Main.kt

```diff
@@ -33,51 +64,36 @@ import kotlinx.coroutines.yield
+    val bootstrapServers = if (args.isNotEmpty()) args[0] else "localhost:29092"
+    val topicName = if (args.size > 1) args[1] else "kkn-serialized-test"
+    logInfo("Main", "Starting test with bootstrapServers: $bootstrapServers and topic: $topicName")
     val producerConfig = mapOf(
-        CommonConfigNames.BOOTSTRAP_SERVERS_CONFIG to "localhost:29092",
+        CommonConfigNames.BOOTSTRAP_SERVERS_CONFIG to bootstrapServers,
         CommonConfigNames.CLIENT_ID_CONFIG to "test-consumer",
         CommonConfigNames.LOG_LEVEL_NATIVE to "7"
     )
-    val producer = KafkaProducer(producerConfig, object : Serializer<String> {
-        override fun serialize(
-            data: String,
-            topic: String?,
-            headers: Iterable<Header>?
-        ): ByteArray? =
-                data.encodeToByteArray()
-    }, object : Serializer<String> {
-        override fun serialize(
-            data: String,
-            topic: String?,
-            headers: Iterable<Header>?
-        ): ByteArray? =
-                data.encodeToByteArray()
-    })
+    val typesMap = mapOf(serializeTypeOf<DiktatSuite>(), serializeTypeOf<DetectSuite>())
+    logDebug("Main", "typesMap = $typesMap")
+    val producer = KafkaProducer(producerConfig, StringSerializer(), JsonSerializer<TaskCore>(typesMap))
     runBlocking {
         launch {
             println("Start consume")
-            val consumerService = KafkaParallelGroupsConsumerService(
+            val consumerService = KafkaParallelConsumerService(
                 ConsumerConfig(
-                    listOf("kkn-parallel-test"),
+                    listOf(topicName),
                     mapOf(
-                        CommonConfigNames.BOOTSTRAP_SERVERS_CONFIG to "localhost:29092",
+                        CommonConfigNames.BOOTSTRAP_SERVERS_CONFIG to bootstrapServers,
                         CommonConfigNames.CLIENT_ID_CONFIG to "test-consumer",
                         CommonConfigNames.GROUP_ID_CONFIG to "test-consumer-group",
                         CommonConfigNames.LOG_LEVEL_NATIVE to "7",
                         ConsumerConfigNames.ENABLE_AUTO_COMMIT_CONFIG to "false",
                         ConsumerConfigNames.AUTO_OFFSET_RESET_CONFIG to "earliest"
 
                     ),
-                    object : Deserializer<String> {
-                        override fun deserialize(data: ByteArray, topic: String?, headers: Iterable<Header>?): String =
-                                data.decodeToString()
-                    },
-                    object : Deserializer<String> {
-                        override fun deserialize(data: ByteArray, topic: String?, headers: Iterable<Header>?): String =
-                                data.decodeToString()
-                    },
-                    object : ConsumerRecordHandler<String, String> {
-                        override fun handle(record: ConsumerRecord<String, String>) {
+                    StringDeserializer(),
```

Comment:
I do not understand this logic. So you need both deserialiser of string and then deserializer of json?

Reply:
yes, StringDeserializer user for key deserialization as a rule